### ISSUE TITLE
TypeError: req.setNoDelay is not a function

### ISSUE THREAD

**Issue by pieterbeulque** [visionmedia/superagent#1331]
This package is being used as a dependency for Auth0 v9 and it throws a `req.setNoDelay is not a function` error when calling this method: [changePassword](https://github.com/auth0/auth0.js/blob/master/src/authentication/db-connection.js#L80)

 Thanks ✌️

**Comment by kornelski** [comment]
That's surprising and I don't see from the code how could it happen, unless something patched superagent in a broken way:

```js

  // initiate request
  const mod = exports.protocols[url.protocol]; // it's require('http') or require('https')

  // request
  const req = (this.req = mod.request(options));

  // set tcp no delay
  req.setNoDelay(true); // Node supports this since forever
```

**Comment by pieterbeulque** [comment]
Hi @kornelski, this keeps popping up as an issue for us. Below are a few things that might help point in the right direction.

Superagent is being imported as a dependency for [Auth0.js](https://github.com/auth0/auth0-js).

This particular project is intended for the browser and is bundled using [JSPM 0.16](https://github.com/jspm/jspm-cli).

JSPM provides support for `require('http')` and `require('https')` using these packages.

 - https://github.com/jspm/nodelibs-http
 - https://github.com/jspm/nodelibs-https
 - https://www.npmjs.com/package/http-browserify

These packages don't look 1:1 compatible with the native modules.

Do you think this might be the issue? I don't see why Superagent would `require('http')` in a browser environment. I think this is accounted for in the package (the `browser` field in `package.json`)?

**Comment by kornelski** [comment]
`req.setNoDelay()` is used only in the Node-only version of superagent. Perhaps instead of packaging the browser-compatible version of superagent, your build system bundles the Node version and then tries to emulate Node.js in the browser?

**Comment by pieterbeulque** [comment]
I was looking in that direction too, yes.

However, in the JSPM docs I found this regarding [Node.js usage](https://github.com/jspm/jspm-cli/blob/master/docs/nodejs-usage.md):

> Packages with a browser field will always use that browser mapping even in Node (https://github.com/jspm/npm/issues/32), pending conditional loading support in SystemJS.

So instead of always picking up on the `browser` key, it looks like it doesn't pick up on it at all.

Any advice on how I can test or verify this?

**Comment by kornelski** [comment]
See if there's a string `"This is not supported in browser version of superagent"` in the compiled bundle.

The node version contains `node-superagent` in the bundle.

**Comment by pieterbeulque** [comment]
Hey, I verified & that string is present in the bundle.

It looks like our build tool is using the `./lib/client.js` file as an entry point, as it should.

**Comment by roy09** [comment]
Was there any resolution to this?

Are we suppose to install with `jspm install superagent -o "{ main: null }"`?

**Comment by kornelski** [comment]
From linked issues it seems this bug has been fixed in jspm 0.17. Make sure you have an up-to-date version of jspm.
